turbopack/crates/turbopack-nft/src/nft.rs

```diff
@@ -0,0 +1,150 @@
+
+pub async fn node_file_trace(
+    project_root: RcStr,
+    input: RcStr,
+    graph: bool,
+    show_issues: bool,
+) -> Result<()> {
+    let op = node_file_trace_operation(project_root.clone(), input.clone(), graph);
+    op.resolve_strongly_consistent().await?;
+
+    if show_issues {
+        let issue_reporter: Vc<Box<dyn IssueReporter>> =
+            Vc::upcast(ConsoleUi::new(TransientInstance::new(LogOptions {
+                project_dir: PathBuf::from(project_root),
+                current_dir: current_dir().unwrap(),
+                show_all: true,
+                log_detail: false,
+                log_level: IssueSeverity::Hint,
+            })));
+
+        handle_issues(op, issue_reporter, IssueSeverity::Error, None, None).await?;
+    }
+
+    Ok(())
+}
+
+#[turbo_tasks::function(operation)]
+async fn node_file_trace_operation(project_root: RcStr, input: RcStr, graph: bool) -> Result<()> {
+    let workspace_fs: Vc<Box<dyn FileSystem>> = Vc::upcast(DiskFileSystem::new(
+        rcstr!("workspace"),
+        project_root.clone(),
+    ));
+    let input_dir = workspace_fs.root().owned().await?;
+    let input = input_dir.join(&format!("{input}"))?;
+
+    let source = FileSource::new(input);
+    let environment = Environment::new(ExecutionEnvironment::NodeJsLambda(
+        NodeJsEnvironment::default().resolved_cell(),
+    ));
+    let module_asset_context = ModuleAssetContext::new(
+        Default::default(),
+        // TODO These test cases should move into the `node-file-trace` crate and use the same
+        // config.
+        // It's easy to make a mistake here as this should match the config in the binary from
+        // turbopack/crates/turbopack/src/lib.rs
+        CompileTimeInfo::new(environment),
+        ModuleOptionsContext {
+            ecmascript: EcmascriptOptionsContext {
+                ..Default::default()
+            },
+            css: CssOptionsContext {
+                enable_raw_css: true,
+                ..Default::default()
+            },
+            // Environment is not passed in order to avoid downleveling JS / CSS for
+            // node-file-trace.
+            environment: None,
+            ..Default::default()
+        }
+        .cell(),
+        ResolveOptionsContext {
+            enable_node_native_modules: true,
+            enable_node_modules: Some(input_dir.clone()),
+            custom_conditions: vec![rcstr!("node")],
+            loose_errors: true,
+            ..Default::default()
+        }
+        .cell(),
+        Layer::new(rcstr!("test")),
+    );
+    let module = module_asset_context
+        .process(Vc::upcast(source), ReferenceType::Undefined)
+        .module();
+
+    let asset = TracedAsset::new(module).to_resolved().await?;
+
+    if graph {
+        print_graph(ResolvedVc::upcast(asset)).await?;
+    } else {
+        println!("FILELIST:");
+        print_list(ResolvedVc::upcast(asset)).await?;
+    }
```

Comment:
would it be more correct to do the printing outside of the turbo task.  Obviously with invalidations disabled we shouldn't expect this to get executed multiple times, but im not sure how much of a guarantee we would have